stdlib/public/core/StaticBigInt.swift

```diff
@@ -0,0 +1,203 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// An immutable arbitrary-precision signed integer.
+///
+/// `StaticBigInt` is primarily intended to be used as the associated type of an
+/// `ExpressibleByIntegerLiteral` conformance.
+///
+///     extension UInt256: ExpressibleByIntegerLiteral {
+///         public init(integerLiteral value: StaticBigInt) {
+///             precondition(
+///                 value.signum() >= 0 && value.bitWidth <= 1 + Self.bitWidth,
+///                 "integer overflow: '\(value)' as '\(Self.self)'"
+///             )
+///             self.words = Words()
+///             for wordIndex in 0..<Words.count {
+///                 self.words[wordIndex] = value[wordIndex]
+///             }
+///         }
+///     }
+@available(SwiftStdlib 5.8, *)
+@frozen
+public struct StaticBigInt:
+  _ExpressibleByBuiltinIntegerLiteral,
+  ExpressibleByIntegerLiteral,
+  Sendable
+{
+  @available(SwiftStdlib 5.8, *)
+  @usableFromInline
+  internal let _value: Builtin.IntLiteral
+
+  @available(SwiftStdlib 5.8, *)
+  @inlinable
+  public init(_builtinIntegerLiteral value: Builtin.IntLiteral) {
+    _value = value
+  }
+
+  /// Returns the given value unchanged.
+  @_alwaysEmitIntoClient
+  @inline(__always)
+  public static prefix func + (_ rhs: Self) -> Self {
+    rhs
+  }
+}
+
+//===----------------------------------------------------------------------===//
+// MARK: - Binary Representation
+//===----------------------------------------------------------------------===//
+
+@available(SwiftStdlib 5.8, *)
+extension StaticBigInt {
+
+  /// Returns `-1`, `0`, or `+1` to indicate whether this value is less than,
+  /// equal to, or greater than zero.
+  ///
+  /// The return type is `Int` rather than `Self`.
```

Comment:
Missing a `- Returns:` here — we could move some of the detail could move down from the abstract.  For one approach, see [Int.signum](https://developer.apple.com/documentation/swift/int/signum()).

Reply:
I'll remove the second paragraph. Can the `- Returns:` be omitted, if only the abstract paragraph remains?

Reply:
The guidelines for writing reference require a `- Returns` for every method that returns a value. Sometimes, like here, there isn't much to say. But keeping the content consistent allows for consistent formatting in when reading in Xcode and the developer library and so on.  Here's one option, riffing on `Int.signum`:

Reply:
Shall I mention the word's Latin origin, or add the following example code?

```swift
extension StaticBigInt {

  /// Indicates the value's sign.
  ///
  /// The "signum" (Latin for "sign") can be compared to zero, using relational
  /// operators, when checking for a negative or nonnegative value.
  ///
  ///     StaticBigInt(-42).signum() < 0  //-> true
  ///     StaticBigInt(42).signum() >= 0  //-> true
  ///
  /// - Returns: `-1` if the value is less than zero, `0` if it is equal to
  ///   zero, or `1` if it is greater than zero.
  public func signum() -> Int
}
```

`BinaryInteger.signum()` and `Int.signum()` would need a different use case.

Reply:
The [signum function](https://en.wikipedia.org/wiki/Sign_function) of a number is -1, 0, or +1; it's never some arbitrary negative or positive value.

Reply:
@xwu We could use `value.signum() != -1`, but using `value.signum() >= 0` seems more natural (as a proxy for `value >= 0` if the type was directly `Comparable`).

Reply:
Of course, there's nothing wrong with it, and the spelling is more natural.

My worry is that the subtext of _only_ showing  comparison to zero—and indeed using language such as "...can be compared to zero..."—is that the value _can't or shouldn't_ be compared to `-1` or `1` in the same vein that `sort` currently _shouldn't_ be relied upon to be stable. However, that's not the case here: `signum` _can_ be relied upon _always_ to yield `-1` or `1` for nonzero arguments and, indeed, _that's what the signum function is_.

Reply:
I'll remove the misleading example code, and push an update soon.

Reply:
> The "signum" (Latin for "sign")

I'd omit the mention of Latin, or move it to a note if you think it's important.  It might help some readers remember the name, most developers probably don't need to know this fact.